### PR TITLE
refactor: Remove all `(ERROR) @error` captures

### PR DESCRIPTION
**Pull request by lucario387**
As discussed in #5421, capturing `@error` is inconsistent, requiring
deep nesting (or priority) in order to correctly have red backgrounds to
it.

Some queries has this capture, some don't. For consistency purposes,
removing all of them is more preferable.

For re-enabling error, either add `(ERROR) @error` or `(ERROR _ @error)`
to your custom queries.

### CODE REVIEW

**Review by amaanq — APPROVED** [review]

**Comment by amaanq** [comment]
I'd probably just delete the error stuff instead of commenting it out

**Comment by amaanq** [comment]
I'd probably just delete the error stuff instead of commenting it out

**Comment by lucario387** [comment]
If I were to remove it, I'll only remove `@error` for `(ERROR)` queries, while some niche ones like redundant_escape for git and such, I'll just comment it out

File: queries/gitattributes/highlights.scm [hunk 1]
```diff
@@ -42,12 +42,12 @@
...
-[
-  (pattern_negation)
-  (redundant_escape)
-  (trailing_slash)
-] @error
+; [
+;   (pattern_negation)
+;   (redundant_escape)
+;   (trailing_slash)
+; ] @error
```
Comment (ObserverOfTime) [inline queries/gitattributes/highlights.scm:49]: These are explicitly parsed as errors, similar to some vim syntax files. Do we want to keep them or should I remove them from the grammar?

Comment (ObserverOfTime) [reply queries/gitattributes/highlights.scm:49]: Maybe we could change them to `@text.warning`?

Comment (lucario387) [reply queries/gitattributes/highlights.scm:49]: I'll go with whatever you and clason choose

**Comment by lucario387** [comment]
> I wouldn't remove all @error captures, just (hard) remove all ERROR nodes.

Just did that. Also un-comment @ObserverOfTime's gitattribute nodes

File: queries/rnoweb/highlights.scm
Comment (lucario387) [inline queries/rnoweb/highlights.scm:1]: I don't know whether deleting the file is ok but it only has `(ERROR) @error` capture, so I remove it

**Comment by lucario387** [comment]
I don't particularly see how automatically adding `(ERROR)` pattern will solve anything, when specificity trumps everything (bar priorities). And `ERROR` nodes, as it's designed right now, will always be the least specific possible.

**Comment by clason** [comment]
> I don't particularly see how automatically adding (ERROR) pattern will solve anything, when specificity trumps everything (bar priorities). And ERROR nodes, as it's designed right now, will always be the least specific possible.

It will solve the issue of 
1. nobody wants these for highlighting
2. some (may) want these for (tree-sitter) debugging

For case 2, one would add them with high(est) priority.

**Review by ObserverOfTime — APPROVED** [review]

**Comment by clason** [comment]
Beat you to it ;)

**Pull request merged**